WalletWasabi/Wallets/Wallet.cs

```diff
@@ -410,6 +410,11 @@ private async void IndexDownloader_NewFilterAsync(object? sender, FilterModel fi
 				if (KeyManager.GetBestHeight() < filterModel.Header.Height)
 				{
 					await ProcessFilterModelAsync(filterModel, CancellationToken.None).ConfigureAwait(false);
+					if (KeyManager.GetBestHeight() < filterModel.Header.Height)
+					{
+						var filterHeight = new Height(filterModel.Header.Height);
+						KeyManager.SetBestHeights(filterHeight, filterHeight);
+					}
```

Comment:
LOC412
```
await ProcessFilterModelAsync(filterModel, CancellationToken.None).ConfigureAwait(false);
```
method takes care of setting the best height. So, I think this block is redundant.

Reply:
No @adamPetho, it does it only when the filter matches and that is exactly what @turbolay is changing here. Anyway, this brings the question: should we remove the `KeyManager.SetBestHeight(height);` in `ProcessFilterModelAsync`? it seems as we should but all this is a stateful thing and that means we could break something unexpectedly so, no, I wouldn't do it.